### PR TITLE
Add resume flag as default for rnafusion run

### PR DESCRIPTION
**Pull request by fevac**
## Description

Related to: https://github.com/Clinical-Genomics/cg/issues/1668

### Changed

Default behaviours for `start` and `run`:
- For `cg workflow rnafusion start`: Do not use -resume
- For `cg workflow rnafusion run`: use -resume

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b set_resume_default -a
    ```

### How to test

- [ ] Run `cg workflow rnafusion start -d solidflawn `. Make sure running command doesn't include `-resume`
- [ ] Run `cg workflow rnafusion run -d solidflawn `. Make sure running command includes `-resume`

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/workflow/rnafusion/base.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def run(
             case_id=case_id,
             log=log,
             work_dir=work_dir,
-            resume=resume,
+            resume=not from_start,
```
Comment (henrikstranneheim) [inline cg/cli/workflow/rnafusion/base.py:132]:
```suggestion
            resume=not_from_start,
```
I assume? or perhaps just `resume`

File: cg/meta/workflow/nextflow_common.py [hunk 2]
```diff
@@ -86,13 +86,37 @@ def create_samplesheet_csv(
 
     @classmethod
     def get_verified_arguments_nextflow(
-        cls, case_id: str, log: Path, pipeline: str, root_dir: str
+        cls, case_id: str, pipeline: str, root_dir: str, log: Path, bg: bool, q: bool
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nextflow_common.py:89]: What is `bg` and `q`? Is there a more mnemonic name for these args?

File: cg/meta/workflow/nextflow_common.py [hunk 3]
```diff
@@ -86,13 +86,37 @@ def create_samplesheet_csv(
...
+        resume: bool,
+        profile: bool,
+        with_tower: bool,
+        stub: bool,
+    ) -> dict:
+        """Transforms click argument related to nextflow run that were left empty
+        into defaults constructed with case_id paths."""
+
+        return {
+            "-w": cls.get_workdir_path(case_id, root_dir, work_dir),
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nextflow_common.py:115]: Use named args

File: tests/cli/workflow/rnafusion/test_cli_rnafusion_compound_commands.py [hunk 4]
```diff
@@ -44,13 +44,25 @@ def test_start(
     # GIVEN decompression is not needed
     RnafusionAnalysisAPI.resolve_decompression.return_value = None
 
+    # WHEN ensuring case config exists where it should be stored
+    Path.mkdir(
```
Comment (henrikstranneheim) [inline tests/cli/workflow/rnafusion/test_cli_rnafusion_compound_commands.py:48]: Same comments about this being a fixture as in the other PR

**Review by henrikstranneheim — APPROVED** [review]
👍

File: cg/cli/workflow/rnafusion/base.py (hunk 1 above)
Comment (fevac) [reply cg/cli/workflow/rnafusion/base.py:132]: well, I wanted to get the opposite of `from_start`. Meaning that if running with `from_start` set as True (so, starting from the beginning without resuming), then resume should be False (so the opposite), and the other way around. 

I could indicate it more clearly by adding:

```
    resume = True
    if from_start:
        resume = False
```
or
```
    resume = not from_start
```

and then just leaving:
```suggestion
            resume=resume,
```

Let me know what you prefer @henrikstranneheim

File: cg/meta/workflow/nextflow_common.py (hunk 2 above)
Comment (fevac) [reply cg/meta/workflow/nextflow_common.py:107]: good point! 
I can change `q` for `quiet`. `bg` needs to stay like that and is for "executing in the background"

File: cg/cli/workflow/rnafusion/base.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/cli/workflow/rnafusion/base.py:132]: I think this looks clear.

```
 resume = True
    if from_start:
        resume = False
```

or even shorter:
```
 resume = False if from_start else True
```

File: cg/meta/workflow/nextflow_common.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg/meta/workflow/nextflow_common.py:107]: How about: `use_child_process` or `use_background_process` at least for the python arg. I do not know if nextflow has a long option for `bg`. In code always use the long option as it is more clear when reading/executing.

File: cg/cli/workflow/rnafusion/base.py (hunk 1 above)
Comment (fevac) [reply cg/cli/workflow/rnafusion/base.py:132]: 🤩

**Comment by fevac** [comment]
Tests output:

- [x] Run includes -resume

```
$ cg workflow rnafusion run -d solidfawn
Case solidfawn exists in status db
Running RNAFUSION analysis for solidfawn
Running command export NXF_PID_FILE=/home/proj/stage/rnafusion/cases/solidfawn/solidfawn_nextflow.pid; cd /home/proj/stage/rnafusion/launch_dir; /home/proj/stage/bin/miniconda3/bin/conda run --name S_rnafusion /home/proj/stage/bin/miniconda3/envs/S_rnafusion/bin/nextflow -bg -quiet -log /home/proj/stage/rnafusion/cases/solidfawn/solidfawn_rnafusion_nextflow_log_2023-01-12_14.56.05.log run /home/proj/stage/rnafusion/workflow/rnafusion -w /home/proj/stage/rnafusion/cases/solidfawn/work -resume -profile singularity,hasta,dev_prio --input /home/proj/stage/rnafusion/cases/solidfawn/solidfawn_samplesheet.csv --outdir /home/proj/stage/rnafusion/cases/solidfawn --genomes_base /home/proj/stage/rnafusion/references --trim True --fusioninspector_filter True --all True  > /home/proj/stage/rnafusion/cases/solidfawn/solidfawn-stdout.log 2> /home/proj/stage/rnafusion/cases/solidfawn/solidfawn-stdout.err < /dev/null &
Dry run: process call will not be executed!!
Did not run analysis: dry-run
```

- [x] Start does not include -resume

```
$ cg workflow rnafusion start -d solidfawn
...
Samplesheet written
Case solidfawn exists in status db
Running RNAFUSION analysis for solidfawn
Running command export NXF_PID_FILE=/home/proj/stage/rnafusion/cases/solidfawn/solidfawn_nextflow.pid; cd /home/proj/stage/rnafusion/launch_dir; /home/proj/stage/bin/miniconda3/bin/conda run --name S_rnafusion /home/proj/stage/bin/miniconda3/envs/S_rnafusion/bin/nextflow -bg -quiet -log /home/proj/stage/rnafusion/cases/solidfawn/solidfawn_rnafusion_nextflow_log_2023-01-12_14.49.09.log run /home/proj/stage/rnafusion/workflow/rnafusion -w /home/proj/stage/rnafusion/cases/solidfawn/work -profile singularity,hasta,dev_prio --input /home/proj/stage/rnafusion/cases/solidfawn/solidfawn_samplesheet.csv --outdir /home/proj/stage/rnafusion/cases/solidfawn --genomes_base /home/proj/stage/rnafusion/references --trim True --fusioninspector_filter True --all True  > /home/proj/stage/rnafusion/cases/solidfawn/solidfawn-stdout.log 2> /home/proj/stage/rnafusion/cases/solidfawn/solidfawn-stdout.err < /dev/null &
Dry run: process call will not be executed!!
Did not run analysis: dry-run
```

**Pull request merged**